index.ts

```diff
@@ -0,0 +1,39 @@
+function fizzbuzz() {
+    for (let i = 1; i < 256; i++) {
+
+        let output: string[] = []
+        let rules = {3: "Fizz", 5: "Buzz", 7: "Bang", 11: "Bong", 13: "Fezz"}
+
+        const checkDivisor = (i: number, divisor: number): boolean => {
+            return i % divisor === 0
+        }
+
+        const applyRule = (i: number, divisor: number) => {
+            if (checkDivisor(i, divisor)) {
+                output.push(rules[divisor])
+            }
+        }
+
+        if (checkDivisor(i, 11)) {
+            applyRule(i, 13)
+            applyRule(i, 11)
+        }
+    
+        if (output.length === 0) {
+            applyRule(i, 3)
+            applyRule(i, 13)
+            applyRule(i, 5)
+            applyRule(i, 7)
+        }
+
+        if (output.length === 0) {
+            console.log(i)
+        } else if (checkDivisor(i, 17)) {
+            console.log(output.reverse().join(""))
+        } else {
+            console.log(output.join(""))
```

Comment:
Once you remove the rule from this, you can use my favourite operator the `ternary operator` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator

Its a shorthand for `if <condition> then <value> else <different value>`
In this case you can do `outputString = output.length === 0 ? output.join("") : i`